persistent/tests/test_persistence.py

```diff
@@ -166,6 +228,14 @@ def _test():
             inst._p_oid = new_OID
         self.assertRaises(ValueError, _test)
 
+    def test_assign_p_oid_not_in_cache_allowed(self):
+        jar = self._makeJar()
+        inst = self._makeOne()
+        inst._p_jar = jar
+        inst._p_oid = 1 # anything goes
+        inst._p_oid = 42
+        self.assertEqual(inst._p_oid, 42)
```

Comment:
> (True, the picklecache has that requirement, but it's arguable that the picklecache is ZODB or storage specific.)

How can you make practical use of `Persistent` without also using `picklecache`?  The whole point of the class is to have changes to an object get registered with the cache.

> De-facto, the documentation is incorrect

Relying on the fact that the C implementation does not enforce a constraint that is explicit in the interface specification is literally "undocumented behavior" (as with a compiler which may or may not enforce some corner of a language spec tightly).  Code which does this is off-roading.  I could see noting this in a version bump, with a prominent mention on the changelog, if we chose to enforce it.